Docs/uty 1764 (#748)
spatialos/gdk-for-unity: docs/content/mobile/android/local-deploy.md

## Prepare your project to connect to a local deployment {#prepare}

To connect your Android device to a local deployment, you need to prepare its configuration. Follow the steps below to ensure any local deployment you launch via your Unity Editor is correctly set up.

1. Open your project in your Unity Editor.
1. Navigate to **SpatialOS** > **GDK Tools configuration** to open the configuration window.
1. In the **Runtime IP for local deployment** field, enter your local machine's IP address. (You can find how to do this on the [Lifehacker website](https://lifehacker.com/5833108/how-to-find-your-local-and-external-ip-address).)
1. Select **Save** and close the window.

> Create and add a MonoBehaviour script to your Android client-worker GameObject. To do this:

Reviewer: this is still the old version, let's try to make them the same for local and cloud as far as possible

Author: The format of adding the MonoBehaviour to the GameObject has been updated on cloud-deploy.md to match the format of local.
The first three steps are similar but cloud then provides more information on adding the authentication token you created to the `DevelopmentAuthToken` field.

1. If your project does not already contain one, create a MonoBehaviour script which inherits from the [`MobileWorkerConnector`](https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/com.improbable.gdk.mobile/Worker/MobileWorkerConnector.cs) and include the functionality you want.

> 1. In your Unity Editor, add the MonoBehaviour script to your Android client-worker GameObject.

Reviewer 2: We talk about these worker GameObjects - do we have this documented anywhere that we can link to?

Reviewer 3: This will be addressed in UTY-1794, but is out of scope of this bugfix.

1. In the **Runtime IP for local deployment** field, enter your local machine's IP address. (You can find how to do this on the [Lifehacker website](https://lifehacker.com/5833108/how-to-find-your-local-and-external-ip-address).)
1. Select **Save** and close the window.

> Create and add a MonoBehaviour script to your Android client-worker GameObject.

Reviewer: this doesn't flow that nicely as it breaks the enumeration. Maybe just remove this sentence?

Reviewer 4: Is this section part of setting up your Unity Editor correctly?
I think it is a separate section, so probably needs this bit of text. Maybe add, "To do this:" to make it clearer.

Author: This is a separate section, which sort of falls under the same category needed for a user to prepare their project to connect to a local deployment. Adding, "To do this:" makes this clearer but also helps separates the steps from the previous section to avoid any confusion between the two.

> Create and add a MonoBehaviour script to your Android client-worker GameObject.

Reviewer 4: ```suggestion
Create and add a MonoBehaviour script to your Android client-worker GameObject. To do this:
```

1. If your project does not already contain one, create a MonoBehaviour script which inherits from the [`MobileWorkerConnector`](https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/com.improbable.gdk.mobile/Worker/MobileWorkerConnector.cs) and include the functionality you want.

> 1. In the Unity Editor, add the MonoBehaviour script to your Android client-worker GameObject. In the FPS Starter Project, navigate to **Assets** > **Fps** > **Prefabs** > **AndroidClientWorker**.

Reviewer 4: Same question as above.

Author: I agree with this, this example may be more confusing than helpful as it may direct the user to a different flow. I'll remove the following part:
`In the FPS Starter Project, navigate to **Assets** > **Fps** > **Prefabs** > **AndroidClientWorker**.`

1. Select **Save** and close the window.

## Unity Editor{#in-editor}
1. If your project does not already contain one, create a MonoBehaviour script which inherits from the [`MobileWorkerConnector`](https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/com.improbable.gdk.mobile/Worker/MobileWorkerConnector.cs) and include the functionality you want. <br> 

> In your Unity Editor, add this script it to your Android client-worker GameObject. In the FPS Starter Project this is located at **Assets** > **Fps** > **Prefabs** > **AndroidClientWorker**.

Reviewer: add this as a separate step. 
1. create monobehaviour
2. add it to your game object

Author: This looks good. Two separate steps reads slightly better imo.

> In your Unity Editor, add this script it to your Android client-worker GameObject. In the FPS Starter Project this is located at **Assets** > **Fps** > **Prefabs** > **AndroidClientWorker**.

Reviewer: ```suggestion
In your Unity Editor, add this script to your Android client-worker GameObject. In the FPS Starter Project, navigate to **Assets** > **Fps** > **Prefabs** > **AndroidClientWorker**.
```

Author: Included this change.
Also, changed "In your Unity Editor" to "In the Unity Editor" to match the same language used elsewhere in the page.

Reviewer: good call! unfortunately I'll have to ask you to change it to "In your Unity Editor" everywhere on the page. We are trying to use "your" everywhere, but missed some places :(

Author: All references to "in the Unity Editor" have been replaced with "in your Unity Editor".

> 1. In your Unity Editor, in the Inspector window, in the script’s drop-down window, ensure that the checkbox `ShouldConnectLocally` is checked.

Reviewer: this part reads weird as we have three location statements `In your Unity Editor, in the Inspector window, in the script’s drop-down window,` @ElleEss  any suggestions?

Author: I reworded this a little bit to the following:
1. In the Unity Editor, navigate to your Android client-worker GameObject and ensure the `ShouldConnectLocally` checkbox is checked in the script's drop-down window of the Inspector window.

1. In the **Editor Settings** window's **Unity Remote** section, set the **Device** setting to **Any Android Device**.
1. Open the Scene that starts both your [client-workers]({{urlRoot}}/content/glossary#client-worker) and [server-workers]({{urlRoot}}/content/glossary#server-worker).
1. On your mobile device, open the **Unity Remote** app. Make sure you allow it permissions for location and camera.

> 1. If your project does not already contain one, create a MonoBehaviour script which inherits from the [`MobileWorkerConnector`](https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/com.improbable.gdk.mobile/Worker/MobileWorkerConnector.cs) and include the functionality you want. <br> 

Reviewer: as these steps are the same for all three of them: should we move it up to "Prepare your project to connect to a local deployment" ?

Author: Agree with this. Adding the information under the "Prepare your project to connect to a local deployment" section makes the page cleaner and the steps easier to read.

Adding the following under Prepare:

Create and add a MonoBehaviour script to your Android client-worker GameObject.

1. If your project does not already contain one, create a MonoBehaviour script which inherits from the [`MobileWorkerConnector`](https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/com.improbable.gdk.mobile/Worker/MobileWorkerConnector.cs) and include the functionality you want.
1. In the Unity Editor, add the MonoBehaviour script to your Android client-worker GameObject. In the FPS Starter Project, navigate to **Assets** > **Fps** > **Prefabs** > **AndroidClientWorker**.
1. In the Unity Editor, navigate to your Android client-worker GameObject and ensure the `ShouldConnectLocally` checkbox is checked in the script's drop-down window of the Inspector window.
